noxfile.py

```diff
@@ -52,9 +54,57 @@ def benchmark_model(session: nox.Session, subpackage: ModelSubpackagePath) -> No
     """Run pytest benchmarks for selected icon4py model subpackages."""
     _install_session_venv(session, extras=["dace", "io", "testing"], groups=["test"])
 
+    results_json_path = os.path.abspath(f"results_{subpackage.replace('/', '_')}.json")
     with session.chdir(f"model/{subpackage}"):
-        session.run(*"pytest -sv --benchmark-only".split(), *session.posargs)
+        session.run(
+            *f"pytest \
+            -v \
+            --benchmark-only \
+            --benchmark-warmup=on \
+            --benchmark-warmup-iterations=30 \
+            --benchmark-json={results_json_path}".split(),
+            *session.posargs,
+            success_codes=[0, NO_TESTS_COLLECTED_EXIT_CODE],
+        )
 
+@nox.session(python=["3.10", "3.11"],
+             requires=["benchmark_model-{python}" + f"({subpackage.id})" for subpackage in MODEL_SUBPACKAGE_PATHS])
+def bencher_baseline(session: nox.Session) -> None:
+    """Run pytest benchmarks and upload them using Bencher (https://bencher.dev/) (cloud or self-hosted)."""
+    bencher_json_file_name = f"merged_benchmark_results_{session.python}.json"
+    merged_results = {"benchmarks": []}
+    files = glob.glob("results_*.json")
+    for file in files:
+        try:
+            with open(file, "r") as f:
+                data = json.load(f)
+                merged_results["benchmarks"].extend(data["benchmarks"])
+                # preserve the first file's metadata for a valid pytest-benchmark json file
+                for key in data.keys():
+                    if key != "benchmarks" and key not in merged_results:
+                        merged_results[key] = data[key]
+        except:
+            # Empty file, i.e. no benchmarks
+            continue
+    with open(bencher_json_file_name, "w") as f:
+        json.dump(merged_results, f, indent=4)
```

Comment:
What's the reason for merging these files instead of uploading separately? Then we would only `requires` the specific subproject, which seems simpler and more in the style of the rest of the nox setup.

Reply:
By gathering all the reports in a unified one, I create less clutter in the Bencher Cloud Dashboard. Otherwise, it will be very messy for the user to navigate through all the reports. I prefer this way.